packages/dds/tree/src/test/utils.ts

```diff
@@ -596,15 +597,22 @@ export function validateSnapshotConsistency(
 	idDifferentiator: string | undefined = undefined,
 ): void {
 	assert.deepEqual(
-		treeA.tree,
-		treeB.tree,
+		prepareTreeForCompare(treeA.tree),
+		prepareTreeForCompare(treeB.tree),
 		`Inconsistent document tree json representation: ${idDifferentiator}`,
 	);
+
 	// Note: removed trees are not currently GCed, which allows us to expect that all clients should share the same
```

Comment:
Nit (while we're here, if you want)
```suggestion
	// Note: removed trees are not currently garbage collected, which allows us to expect that all clients should share the same
```